src/components/dialogs/generator-modification-dialog.js

```diff
@@ -141,14 +141,15 @@ const GeneratorModificationDialog = ({
           })
         : undefined;
 
-    const [energySource, energySourceField] = useEnumValue({
+    const [energySource, energySourceField] = useOptionalEnumValue({
         label: 'EnergySourceText',
         inputForm: inputForm,
         formProps: filledTextField,
-        enumValues: ENERGY_SOURCES,
-        defaultValue: getValue(formValues?.energySource),
+        enumObjects: ENERGY_SOURCES,
+        defaultValue: formValues?.energySource
+            ? formValues.energySource.value
+            : null,
```

Comment:
i think you can make condition like this    formValues?.energySource?.value ?? null

Reply:
ok